app/src/main/java/com/example/calorietracker/FirstPageActivity.java

```diff
@@ -0,0 +1,42 @@
+package com.example.calorietracker;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+public class FirstPageActivity extends AppCompatActivity {
+    private FirebaseAuth mAuth;
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        mAuth = FirebaseAuth.getInstance();
+        FirebaseUser currentUser = mAuth.getCurrentUser();
+        if (currentUser != null) {
+            reload(new DashboardActivity());
+        }
+    }
+
+    private void reload(Activity activity) {
+        startActivity(new Intent(FirstPageActivity.this, activity.getClass()));
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_first_page);
+
+        TextView textView = findViewById(R.id.textView5);
+        textView.setOnClickListener(text -> openPersonalDataActivity());
+    }
+
+    public void openPersonalDataActivity() {
+        startActivity(new Intent(FirstPageActivity.this, StartActivity.class));
```

Comment:
Deci de fapt FirstPageActivity e un fel de activitate care te muta din login in start? Ca daca da, nu are sens. Poti adauga logica asta in StartActivity.

Reply:
StartActivity e pagina de unde selecteaza user-ul daca vrea Login sau Register, iar FirstPageActivity este MainActivity